functions.php

```diff
@@ -33,12 +33,20 @@ function createRecipeCards(array $recipes): string {
         if (count($recipes)>0) {
             $recipeCards = '';
             foreach ($recipes as $recipe) {
-                $recipeCards .= "<section class= 'recipe_card'>";
+                $recipeCards .= "<section class='recipe_card'>";
                 $recipeCards .= "<h2>{$recipe['recipe']}</h2>";
                 $recipeCards .= "<h3>Cuisine: {$recipe['cuisine']}</h3>";
                 $recipeCards .= "<h3>Time (mins): {$recipe['time']}</h3>";
                 $recipeCards .= "<a href= {$recipe['link']}>Link to recipe</a>";
-                $recipeCards .= "</section>";
+                $recipeCards .= '</section>';
+                $recipeCards .= '<form action="edit.php" method="post">';
```

Comment:
Weird having the form tags generate in the function. Could make it hard to spot html errors. Having the function just generate inputs might be good for separation of concerns :)

Reply:
Might wait and see what Ash says on this one - I couldn't have inputs outside the form tags and I want it to repeat for each recipe so without combining html and php I'm not sure how it would work :)